db/version_set.h

```diff
@@ -130,6 +130,32 @@ class VersionStorageInfo {
 
   void AddFile(int level, FileMetaData* f);
 
+  // Resize/Initialize the space for compact_cursors_by_level_
+  void ResizeCompactCursors(int level) {
+    compact_cursors_by_level_.resize(level, InternalKey());
+  }
+
+  const std::vector<InternalKey>& GetLeveledCompactCursors() const {
+    return compact_cursors_by_level_;
+  }
+
+  const std::vector<InternalKey>& GetNextLeveledCompactCursors() const {
+    return next_compact_cursors_by_level_;
+  }
+
+  // REQUIRES: ResizeCompactCursors has been called
+  void AddCursorForOneLevel(int level, const InternalKey& largest_compact_key) {
+    compact_cursors_by_level_[level] = largest_compact_key;
+  }
+
+  // Called when a compaction succeeds
+  void UpdateCompactCursors(const ImmutableOptions& immutable_options,
+                            int level) {
+    if (immutable_options.compaction_pri == kRoundRobin) {
```

Comment:
Any other condition need to be considered here? How to determine if the current compaction job is the one that used RoundRobin policy?